app/models/main.py

```diff
@@ -1093,15 +1096,24 @@ class DraftService(db.Model, ServiceTableMixin):
                            default=None)
 
     @staticmethod
-    def from_service(service):
-        return DraftService(
-            framework_id=service.framework_id,
-            lot=service.lot,
-            service_id=service.service_id,
-            supplier=service.supplier,
-            data=service.data,
-            status=service.status
-        )
+    def from_service(service, questions_to_copy=None, target_framework_id=None):
+        draft_not_on_same_framework_as_service = target_framework_id and target_framework_id != service.framework.id
+
+        kwargs = {
+            'framework_id': service.framework_id if not target_framework_id else target_framework_id,
+            'lot': service.lot,
+            'service_id': service.service_id,
+            'supplier': service.supplier,
+            'data': {
+                key: value for key, value in service.data.items() if key in questions_to_copy
+            } if draft_not_on_same_framework_as_service else service.data,
```

Comment:
If we specify 'target_framework_id' but don't specify 'questions_to_copy' this will break in a weird way (doing an `in` check on `None`. Should probably firm this up a bit to have a check early on that `questions_to_copy` is a list of strings if `draft_not_on_same_framework_as_service` is True.

Reply:
Cool. I've added some logic, and also some tests to make sure this is doing what it should.